src/calliope/model.py

```diff
@@ -302,26 +302,62 @@ def _add_math(self, add_math: list) -> AttrDict:
             if not yaml_filepath.is_file():
                 file_errors.append(filename)
                 continue
-            else:
-                override_dict = AttrDict.from_yaml(yaml_filepath)
+            math.union(AttrDict.from_yaml(yaml_filepath), allow_override=True)
 
-            base_math.union(override_dict, allow_override=True)
         if file_errors:
             raise exceptions.ModelError(
                 f"Attempted to load additional math that does not exist: {file_errors}"
             )
-        self._model_data.attrs["applied_additional_math"] = add_math
-        return base_math
+        return (math, applied_math)
+
+    def _math_update_with_mode(
```

Comment:
This is the function I removed from the backend. Works in the same way, except that it returns the updated math instead of directly modifying it.

Also, I made sure to avoid the use of `.copy()` unless absolutely needed. Should be more efficient in cases were no modification is necessary.